src/utils.ts

```diff
@@ -314,14 +316,24 @@ export async function renderAbstractUnimplementedFunctions(contract: ContractDef
   const currentSelectors = [...contract.vStateVariables, ...contract.vFunctions].map((node) => node.raw?.functionSelector);
   const inheritedSelectors = getAllInheritedSelectors(contract);
 
+  // If the abstract contract has a constructor, we need to add it to the selectors
+  if (contract.vConstructor) {
+    const constructors = inheritedSelectors['constructor'];
+    inheritedSelectors['constructor'] = {
+      implemented: constructors.implemented,
+      contracts: constructors.contracts ? constructors.contracts.add(contract.name) : new Set([contract.name]),
+      functions: constructors.functions ? [...constructors.functions, contract.vConstructor] : [contract.vConstructor],
+    };
+  }
+
   for (const selector in inheritedSelectors) {
     // Skip the functions that are already implemented in the current contract
     if (currentSelectors.includes(selector)) continue;
 
     // Skip the functions that are already implemented in the inherited contracts
     if (inheritedSelectors[selector].implemented) continue;
 
-    const func = inheritedSelectors[selector].function;
+    const func = inheritedSelectors[selector].functions[0];
```

Comment:
@agusduha could you clarify the changes to this function? In particular, I have 2 questions:
- Why `function` has turned into `functions`
- Why getting the first function here?

Reply:
The thing is that now we want to save all the constructors functions definitions, but we still don't need them for a common function. For functions that are not constructor there is a list with only 1 node. Maybe to make this better we have to separate functions from constructors, anyway we still need to have ona base node to not change the context function.